ini.go

```diff
@@ -40,23 +40,26 @@ func iniLoadOrEmpty(filename string) (*ini.File, error) {
 
 // iniSave safely writes the ini file to the named file.
 func iniSave(filename string, iniFile *ini.File) error {
-	finfo, err := os.Stat(filename)
-	if err != nil {
-		return err
-	}
-	f, err := safefile.Create(filename, finfo.Mode())
+	f, err := safefile.Create(filename, 0644)
```

Comment:
I would calculate the mode at the very beginning, before calling safefile.Create. Your code seems to make use of the fact that safefile.Create will not overwrite the original filename until Close/Commit is called, but it looks more like an error when reading the code (it forces you to know how safefile.Create behaves to understand you are not checking the stats too late).

You can initialize mode = 0644 and uid/gid to -1, and set them to filename actual attributes if it exists, so the code from that point onwards is generic:

```
mode = 0666
uid = -1
gid = -1
if fileExists(filename) {
mode = filename.Mode()
uid = sys.Uid
gid = sys.Gid
}
f = safefile.Create(filename)
// Fixup permissions/ownership
f.Chmod(f.File.Name(), mode)
os.Chown(f.File.Name(), uid, gid)
..
```